icu4c/source/i18n/messageformat2_errors.h

```diff
@@ -66,6 +66,15 @@ namespace message2 {
     enum DynamicErrorType {
         UnresolvedVariable,
         FormattingError,
+        BadOptionError,
+        // This is used to signal errors from :number and :integer when a
+        // bad `select` option is passed. In this case, fallback output
+        // is not used, so it must be distinguished from a regular bad
+        // option error (but it maps to a bad option error in the final
+        // error code).
+        // See https://github.com/unicode-org/message-format-wg/blob/main/spec/functions/number.md#number-selection
+        // ("The formatting of the _resolved value_ is not affected by the `select` option.")
+        SoftBadOptionError,
```

Comment:
docs on SoftBadOption ?

Reply:
Fixed in [e525cac](https://github.com/unicode-org/icu/pull/3423/commits/e525cacb99e2185ec4e6a10965015c84004f80d8)